File: DSA_c_programming/DSA_c_programming/cirq.c
#include<stdio.h>
#include<stdlib.h>

#define max 10
int q[max];
//int p[20];
int f=-1;
int r=-1;
void deq();
void display();
void enq();
int main(){
	int n;
	while(1){
		printf("choose\n1.enq\n2.deq\n3.exit\n4.display\n");
		scanf("%d",&n);
		switch(n){
			case 1:enq();break;
			case 2:deq();break;	
			case 4:display();break;
			case 3:exit(0);
		}
	}
return 0;
}
/*int element;
	printf("enter elemnt");
	scanf("%d",&element);
	if(((rear+1)%MAX)==front)
	{
		printf("queue over fl0w\n");
		return ;
	}
	else if(front ==-1)
		front=rear=0;
	else
	{
		rear=((rear+1)%MAX);
	}
	cqueue[rear]=element;*/
void enq(){
	int data;
	printf("enter data\n");
	scanf("%d",&data);
	if(f==0 && r==max-1){
		printf("overflow\n");
		return ;
	}
	else if(f==-1){
		f=0;
		r=0;
		q[r]=data;
	}
	else{
		r=(r+1)%max;
		q[r]=data;
	}
}
void deq(){
	if(f==-1)
		printf("underflow\n");
	else if(f==r){
		f=-1;
		r=-1;
	}
	else{
		f=(f+1)%max;
	}
}
void display(){
	int i;
	if(f==-1)
	{
		printf("queue is empty\n");
		return;
	}
	for(i=f;i!=r;i=(i+1)%max)
	{
		printf("%d\t",q[i]);
	}
	printf("%d\t",q[i]);
	printf("\n");
}

File: ddd/cirq.java
import java.util.*;
import static java.lang.System.out;

class cirq {
    public static void main(String args[]) {
        int k, x;
        Scanner sc = new Scanner(System.in);
        queue q = new queue();
        while (true) {

            out.printf("enter 1.enq\n2.deq\n3.display\n4.exit\n");
            k = sc.nextInt();
            switch (k) {
                case 1:
                    out.println("enter element to insert\n");
                    x = sc.nextInt();
                    q.enq(x);
                    break;
                case 2:
                    q.deq();
                    break;
                case 3:
                    q.display();
                    break;
                default:
                    return;
            }
        }
    }
}

class queue {
    int front = -1, rear = -1;
    int q[] = new int[5];

    void enq(int x) {
        if (this.front == -1 || this.rear == -1) {
            this.front = 0;
            this.rear = 0;
            q[this.rear] = x;
        } else if (this.front == (this.rear + 1) % 5) {
            out.println("overflow");
        } else {
            this.rear = (this.rear + 1) % 5;
            q[this.rear] = x;
        }
    }

    void deq() {
        if (this.rear == -1 && this.front == -1) {
            out.println("underflow");
        } else if (this.front == this.rear) {
            out.println("the deleted element is " + q[this.front]);
            this.front = -1;
            this.rear = -1;
        } else {
            out.println("the deleted element is " + q[this.front]);
            this.front = (this.front + 1) % 5;
        }
    }

    void display() {
        int i = this.front;
        if (this.front == -1) {
            out.println("empty");
            return;
        }
        while (i != this.rear) {
            out.printf("%d  ", q[i]);
            i = (i + 1) % 5;
        }
        out.println(q[this.rear]);
    }
}